Automattic/wp-calypso
# Refactor Notice component to functional Typescript

#### Changes proposed in this Pull Request

In an effort to decouple the `Notices` component from calypso, I am also rewriting the needed components to TS functional components hoping that this will reduce some regressions.

**Todo**

- [ ] Fix the tests https://github.com/Automattic/wp-calypso/blob/f2c9d99b07e14ea938829e9b2e5b31b6fb388a90/client/components/notice/test/index.js#L14

#### Testing instructions



* Inspect /devdocs/design/notice in this live branch and in https://wpcalypso.wordpress.com/devdocs/design/
* Go to posts screen, delete one post and then undo.
* Check for regressions



Related to #51276
